### PR TITLE
centrally define default config file; get absolute path to config file

### PR DESCRIPTION
**Pull request by kamfonik**
This is required for the helpdesk VM setup to run, because the directory where the user is running the commands is different from the directory containing the scripts and settings.ini. 

This also sets up argparse in the addusers and set-quotas scripts, for specifying the appropriate configuration file as an argument.  We'll make additional use of argparse later.

### CODE REVIEW

File: addusers.py [hunk 1]
```diff
@@ -257,6 +246,30 @@ def parse_rows(rows):
...
+    if args.config is not None:
+        CONFIG_FILE = set_config_file(args.config)
+    else:
+        CONFIG_FILE = set_config_file()
+
+    config = ConfigParser.ConfigParser()
+    config.read(CONFIG_FILE)
+
+    admin_user = config.get('auth', 'admin_user')
+    admin_pwd = config.get('auth', 'admin_pwd')
```
Comment (robbaronbu) [inline addusers.py]: It is a little disconcerting that the admin user and password has to be included configuration file.  I am OK with this if it is the only way to do this, however, this probably would not pass a security audit.

Comment (kamfonik) [inline addusers.py]: I don't think there's a simple way around that.  In production the scripts don't authenticate as the actual "admin" user - they use a different user with admin privileges.  Not much better but at least we can disable it if needed without breaking things.

I expect we could lock it down further by creating a Keystone role with permissions for only the API calls the script needs.  But that would take some time and doesn't fix the password in a file problem. 

I've set things up in the helpdesk VM so the helpdesk users won't be able to see this file.

File: addusers.py [hunk 2]
```diff
@@ -257,6 +246,30 @@ def parse_rows(rows):
...
+    
+    help_description = ("Add new users and projects to OpenStack using " 
+                        "data from Google Sheets.")
+    parser = argparse.ArgumentParser(description=help_description)
+    parser.add_argument('-c', '--config', 
+                        help='Specify configuration file.')
+
+    args = parser.parse_args()
+   
+    if args.config is not None:
```
Comment (knikolla) [inline addusers.py]: This conditional is not required. If ``args.config`` is ``None`` you are sending None to ``set_config_file(cfg_file=None)`` which knows how to interpret the ``None`` value.

File: addusers.py [hunk 3]
```diff
@@ -257,6 +246,30 @@ def parse_rows(rows):
...
+        CONFIG_FILE = set_config_file()
+
+    config = ConfigParser.ConfigParser()
+    config.read(CONFIG_FILE)
+
+    admin_user = config.get('auth', 'admin_user')
+    admin_pwd = config.get('auth', 'admin_pwd')
+    admin_project = config.get('auth', 'admin_project')
+    auth_url = config.get('auth', 'auth_url')
+    nova_version = config.get('nova', 'version')
```
Comment (knikolla) [inline addusers.py]: Why not put all these lines in ``config.py`` also? You're reusing them in ``reset-password.py``

Comment (kamfonik) [inline addusers.py]: Yes, this is planned, I had wanted to think about the best way to set up the centralized config info.  Would you recommend defining a config object with these attributes?  It seems you did something like that in the setpass config.py, using oslo_config.

**Comment by xuhang57** [comment]
Oops. Merge conflict @kamfonik

**Comment by kamfonik** [comment]
@xuhang57 yeah, I expected that to happen with these, I will resolve before merging.

**Comment by xuhang57** [comment]
@kamfonik  free feel to ping me if you need help or review. I will work a bit over the weekend so.

**Comment by kamfonik** [comment]
rebased and removed merge conflicts - just a line or two in import statements where the edits overlapped

**Pull request merged**